frontend/server/src/Controllers/Course.php

```diff
@@ -2792,12 +2792,24 @@ public static function getCourseListDetailsForSmarty(\OmegaUp\Request $r): array
 
         $page = (isset($r['page']) ? intval($r['page']) : 1);
         $pageSize = (isset($r['page_size']) ? intval($r['page_size']) : 1000);
+        $coursesTypes = ['public', 'student'];
+
+        \OmegaUp\Validators::validateOptionalInEnum(
+            $r['access_mode'],
+            'access_mode',
+            $coursesTypes
+        );
 
         $courses = self::getCoursesList($r->identity, $page, $pageSize);
-        $coursesTypes = ['public', 'student'];
 
         $filteredCourses = [];
         foreach ($coursesTypes as $courseType) {
+            if (
+                $r['access_mode'] != $courseType
+                && !is_null($r['access_mode'])
```

Comment:
nit:

```suggestion
                !is_null($r['access_mode']) &&
                $r['access_mode'] != $courseType
```

es mejor si se hace la validación al revés.

Reply:
Listo, aunque esto está medio raro. Si invierto las condiciones así como indicas, obtengo este error:
```
ERROR: RedundantCondition - frontend/server/src/Controllers/Course.php:2808:17 - Found a redundant condition when evaluating $courseType and trying to reconcile type 'string(public)|string(student)' to !string(public)|string(student) (see https://psalm.dev/122)
                !is_null($r['course_type']) &&
                $r['course_type'] != $courseType
```
Y si suena algo redundante. Pero curiosamente si se requieren las dos condiciones, porque `courseType` puede tener esos tres valores: 'public', 'student' o `null`. ¿Cómo le puedo hacer para arreglar eso?

Reply:
Ahhh, creo que se puede arreglar con lo que mencionas en este comentario: https://github.com/omegaup/omegaup/pull/4424#discussion_r460441039.
Deja aplicar esos cambios y ya no tengo que sufrir con este problema.

Reply:
Listo, esto ya quedó